### ISSUE TITLE
FR: Create a way to increase the time between frames.

### ISSUE THREAD

**Issue by apwilson** [flutter/flutter#7545]
Unlike timeDilation which doesn't change the frame rate but slows down any ticking animations (or something like that), this feature would slow down how often frames are submitted (or something like that).

Use cases:

1. If a developer has a weird single frame jump they could use this to 'see' that frame better to aid in debugging.
1. If a developer relies on sizes from a previous frame in their app (which is bad in general) this would increase the jank caused by that to possible 'real world' levels.

**Comment by abarth** [comment]
We could make the scheduler drop N frames before actually pushing a frame.

**Comment by abarth** [comment]
Would you want to do this in conjunction with time dilation so the animations would proceed with roughly the same number of frames (just spread out over a longer time period)?

**Comment by apwilson** [comment]
Yeah possibly.  I think they are independent things - two different knobs to tweak when debugging.

**Comment by apwilson** [comment]
I don' t think 'drop N frames' is the API I'd want.  I think I'd want 'debugMaxFps' or 'debugMinDurationBetweenFrames' as the API to set.

**Comment by HansMuller** [comment]
Maybe we need a debug way to manually step through frames?

**Comment by apwilson** [comment]
Yeah, that sounds good too.

**Comment by dnfield** [comment]
Do we still want this given the timeline api?  It seems like we should be able to just very cheaply inject timeline events with arbitrary parameters and see what frames are causing problems that way.  From what I can tell, this has been the strategy used by team members chasing down performance issues, and it works well because it has minimal impact to the performance of the application as is and doens't require as much guessing about how slow or which frame you want to capture.

**Comment by Piinks** [comment]
This issue is being closed as obsolete. While slowing down frame submission was a useful debugging concept in the early days of the framework, the introduction of the **Flutter DevTools Performance view** and the **Timeline API** provides a much more powerful and non-invasive way to diagnose jank and inspect specific frames. Developers can now record and "scrub" through individual frames to identify performance bottlenecks with high precision.

If there is more work to do here, please let us know by filing a new issue with up to date information. Thanks!
